src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -4,27 +4,36 @@
  */
 package com.vmware.vipclient.i18n.messages.service;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-
-import org.json.simple.JSONArray;
+import java.util.*;
 
 import com.vmware.vipclient.i18n.VIPCfg;
 import com.vmware.vipclient.i18n.base.DataSourceEnum;
+import com.vmware.vipclient.i18n.common.ConstantsMsg;
+import com.vmware.vipclient.i18n.messages.api.opt.ComponentOpt;
 import com.vmware.vipclient.i18n.messages.api.opt.server.ProductBasedOpt;
 import com.vmware.vipclient.i18n.messages.dto.BaseDTO;
+import com.vmware.vipclient.i18n.messages.dto.LocaleDTO;
 import com.vmware.vipclient.i18n.messages.dto.MessagesDTO;
+import com.vmware.vipclient.i18n.util.FormatUtils;
 import com.vmware.vipclient.i18n.util.LocaleUtility;
+import org.json.simple.JSONArray;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ProductService {
-    private MessagesDTO dto = null;
+    private BaseDTO dto = null;
+    Logger logger = LoggerFactory.getLogger(ProductService.class);
 
-    public ProductService(MessagesDTO dto) {
+    public ProductService(BaseDTO dto) {
         this.dto = dto;
     }
 
-    // get supported components defined in vip service
+    /**
+     * get supported components defined in vip service
+     * @return JSONArray
+     * @deprecated Replaced by {@link #getComponents(Iterator<>)}
+     */
+    @Deprecated
```

Comment:
I see you write new method to get componet list, this method isn't exposed to user, why not remove it?